frontend/pages/hosts/details/HostDetailsPage/modals/LockModal/LockModal.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useContext } from "react";
+
+import hostAPI from "services/entities/hosts";
+
+import Modal from "components/Modal";
+import Button from "components/buttons/Button";
+import Checkbox from "components/forms/fields/Checkbox";
+import { NotificationContext } from "context/notification";
+import { AxiosError } from "axios";
+
+const baseClass = "lock-modal";
+
+interface ILockModalProps {
+  id: number;
+  platform: string;
+  hostName: string;
+  onSuccess: () => void;
+  onClose: () => void;
+}
+
+const LockModal = ({
+  id,
+  platform,
+  hostName,
+  onSuccess,
+  onClose,
+}: ILockModalProps) => {
+  const { renderFlash } = useContext(NotificationContext);
+  const [lockChecked, setLockChecked] = React.useState(false);
+  const [isLocking, setIsLocking] = React.useState(false);
+
+  const onLock = async () => {
+    setIsLocking(true);
+    try {
+      await hostAPI.lockHost(id);
+      onSuccess();
+      renderFlash("success", "Host locked successfully!");
+    } catch (error) {
+      const err = error as AxiosError;
+      renderFlash("error", err.message);
```

Comment:
<img width="652" alt="Screenshot 2024-02-12 at 10 48 02 AM" src="https://github.com/fleetdm/fleet/assets/71795832/7c1848f6-f406-4232-8b2d-a49d8984186c">

Are these Figma specs a part of this ticket?

Reply:
yes. those messages should come though the API and the UI will display them